src/Servers/Connections.Abstractions/src/Features/IStreamAbortFeature.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Connections.Features
+{
+    /// <summary>
+    /// Supports aborting one side of a connection stream.
+    /// </summary>
+    public interface IStreamAbortFeature
+    {
+        /// <summary>
+        /// Abort the read side of the connection stream.
+        /// </summary>
+        /// <param name="errorCode">The error code to send with the abort.</param>
+        /// <param name="abortReason">An optional <see cref="ConnectionAbortedException"/> describing the reason to abort the read side of the connection stream.</param>
+        void AbortRead(long errorCode, ConnectionAbortedException abortReason);
+
+        /// <summary>
+        /// Abort the write side of the connection stream.
+        /// </summary>
+        /// <param name="errorCode">The error code to send with the abort.</param>
+        /// <param name="abortReason">An optional <see cref="ConnectionAbortedException"/> describing the reason to abort the write side of the connection stream.</param>
+        void AbortWrite(long errorCode, ConnectionAbortedException abortReason);
```

Comment:
In HTTP/3 the read and write sides of the stream can be aborted independently. An error code is sent with these aborts.

I followed the pattern of `ConnectionContext.Abort(ConnectionAbortedException abortReason)` which uses the exception to pass in a reason. The reason isn't sent to the peer in HTTP/3 and is only used by logging. It can be removed if we don't think passing a reason adds value.

Reply:
Do thumbs up mean keep the exception? 😄